packages/urql-exchange/test/urql-exchange.test.ts

```diff
@@ -0,0 +1,118 @@
+import { createClient, OperationResult } from '@urql/core'
+import { yogaExchange } from '../src'
+import { observableToAsyncIterable } from '@graphql-tools/utils'
+import { pipe, toObservable } from 'wonka'
+import { createServer } from 'graphql-yoga'
+import { File } from '@whatwg-node/fetch'
+
+describe('graphExchange', () => {
+  const port = 4000 + Math.floor(Math.random() * 1000)
+  const endpoint = '/graphql'
+  const hostname = '127.0.0.1'
+  const yoga = createServer({
+    hostname,
+    port,
+    endpoint,
+    logging: false,
+    maskedErrors: false,
+    schema: {
+      typeDefs: /* GraphQL */ `
+        scalar File
+        type Query {
+          hello: String
+        }
+        type Mutation {
+          readFile(file: File!): String!
+        }
+        type Subscription {
+          time: String
+        }
+      `,
+      resolvers: {
+        Query: {
+          hello: () => 'Hello Urql Client!',
+        },
+        Mutation: {
+          readFile: (_, args: { file: File }) => args.file.text(),
+        },
+        Subscription: {
+          time: {
+            async *subscribe() {
+              while (true) {
+                await new Promise((resolve) => setTimeout(resolve, 1000))
+                yield new Date().toISOString()
+              }
+            },
```

Comment:
We don't really need to use setTimeout here, can we change the test to not take "real" time?